src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -1433,3 +1544,70 @@ def merge_metadata(a_metadata, b_metadata, *, source, update=(), min_=(), max_=(
         result_directory=r_a.result_directory,
         app_version_info=app_version_info,
     )
+
+
+@contextmanager
+def open_global_sensitivity_analysis_result_file(
+    results_directory: Path,
+) -> Iterator[Optional[h5py.File]]:
+    """
+    Open a global sensitivity analysis result file.
+    """
+    filename = results_directory / "uq_result"
+    ignored_file = results_directory / "uq_result.creating"
+    # Avoid to read result files with incomplete metadata.
+    if not ignored_file.is_file():
+        with _open_result_file(filename) as file:
+            yield file
+    else:
+        yield
+
+
+def read_global_sensitivity_analysis_meta_data(
+    result_directory: Path,
+) -> Optional[GlobalSensitivityAnalysisMetadata]:
+    """
+    Read the global sensitivity analysis metadata persisted in a result file.
+    """
+
+    return GlobalSensitivityAnalysisMetadata.get_metadata_from_dir(
+        result_directory=result_directory
+    )
+
+
+def read_global_sensitivity_analysis_time_set(
+    result_directory: Path,
+) -> Optional[numpy.array]:
+    """
+    Get the time set for sensitivity analysis results.
+    """
+    with open_global_sensitivity_analysis_result_file(
+        results_directory=result_directory
+    ) as result_file:
+        if not result_file:
+            return
+        return result_file[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME]["time_set"][:]
+
+
+def read_global_sensitivity_coefficients(
+    coefficients_key: str, results_directory: Path
+) -> Optional[Dict[str, numpy.array]]:
+    """
+    Read the global sensitivity analysis coefficients results.
+    """
+    uq_meta_data = read_global_sensitivity_analysis_meta_data(results_directory)
+    with open_global_sensitivity_analysis_result_file(results_directory) as result_file:
+        if not result_file:
+            return
+        global_sensitivity_analysis_coefficients = result_file[
+            GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME
+        ]["global_sensitivity_analysis"][:]
+
+    meta_information = uq_meta_data.global_sensitivity_analysis_meta_data[
+        coefficients_key
+    ]
+    return {
+        coefficients_key: global_sensitivity_analysis_coefficients[
+            meta_information.qoi_index, meta_information.qoi_data_index
+        ]
+    }
```

Comment:
Sorry, this point was bypassing for me. Really we dont need the key. Thanks @ro-oliveira95 and @prusse-martin.

Reply:
If i do as you mentioned, i take this error:
```shell
tests/results/test_aggregator.py:326 (test_read_uq_global_sensitivity_analysis)
global_sa_results_dir = PosixPath('/tmp/pytest-of-gustavopb/pytest-91/test_read_uq_global_sensitivit0/test_aggregator/fake_uq_dir')

    def test_read_uq_global_sensitivity_analysis(global_sa_results_dir: Path) -> None:
>       data = read_global_sensitivity_coefficients(coefficients_key="temperature::parametric_var_1@trend_id_1",results_directory=global_sa_results_dir)

test_aggregator.py:328: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/alfasim_sdk/result_reader/aggregator.py:1557: in read_global_sensitivity_coefficients
    return global_sensitivity_analysis_coefficients[meta.qoi_index,  meta.qoi_data_index]
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
../../../../Work/miniconda/envs.3/alfasim_gui-py310/lib/python3.10/site-packages/h5py/_hl/dataset.py:766: in __getitem__
    if self._fast_read_ok and (new_dtype is None):
../../../../Work/miniconda/envs.3/alfasim_gui-py310/lib/python3.10/site-packages/h5py/_hl/base.py:534: in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
../../../../Work/miniconda/envs.3/alfasim_gui-py310/lib/python3.10/site-packages/h5py/_hl/dataset.py:745: in _fast_read_ok
    self._extent_type == h5s.SIMPLE
../../../../Work/miniconda/envs.3/alfasim_gui-py310/lib/python3.10/site-packages/h5py/_hl/base.py:534: in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
../../../../Work/miniconda/envs.3/alfasim_gui-py310/lib/python3.10/site-packages/h5py/_hl/dataset.py:634: in _extent_type
    return self.id.get_space().get_simple_extent_type()
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: Invalid dataset identifier (invalid dataset identifier)
```
Só to access the an specific data i do this: ```result_file[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME]['global_sensitivity_analysis'][meta.qoi_index,  meta.qoi_data_index]``` and works good.

Reply:
ah... That is because I was indexing the dset after the file has been closed the correct think I should suggest was:

```suggestion
    metadata = read_global_sensitivity_analysis_meta_data(results_directory)
    meta = metadata.global_sensitivity_analysis_meta_data[coefficients_key]
    
    with open_global_sensitivity_analysis_result_file(results_directory) as result_file:
        if not result_file:
            return
        coefficients_dset = result_file[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME][
            "global_sensitivity_analysis"
        ]
        return {
            coefficients_key: coefficients_dset[meta.qoi_index, meta.qoi_data_index]
        }